### PR TITLE
feat(UI): integration extension UI work for ENTESB-13249

### PR DESCRIPTION
**Pull request by kahboom**
Relates to [ENTESB-13249](https://issues.redhat.com/browse/ENTESB-13249) (UI work for Integration Extension library)

Screenshot in Integration Editor final step:

Screenshot 2020-05-06 17 48 04

related:
- https://issues.redhat.com/browse/ENTESB-13249
- https://issues.redhat.com/browse/ENTESB-11534

### CODE REVIEW

File: app/ui-react/packages/api/src/useIntegrationHelpers.tsx [hunk 1]
```diff
@@ -13,11 +13,7 @@ import produce from 'immer';
 import * as React from 'react';
 import { ApiContext } from './ApiContext';
 import { callFetch } from './callFetch';
-import {
-  API_PROVIDER_END_ACTION_ID,
-  PUBLISHED,
-  UNPUBLISHED,
-} from './constants';
+import { API_PROVIDER_END_ACTION_ID, PUBLISHED, UNPUBLISHED, } from './constants';
```
Comment (gashcrumb) [inline app/ui-react/packages/api/src/useIntegrationHelpers.tsx:16]: seems like your IDE doesn't agree with our formatting :-)

File: app/ui-react/packages/api/src/useExtensions.tsx [hunk 2]
```diff
@@ -8,13 +8,28 @@ export interface IExtensionsResponse {
...
+export const useExtensions = (disableUpdates: boolean = false, extensionType?: IExtensionType) => {
+  /**
+   * Here we are allowing calls to `useExtensions` to pass an optional `extensionType`
+   * query param of type IExtensionType, which gets appended to the string URL when
+   * provided.
+   *
+   * We currently use this for listing available Integration Extensions in the final
+   * step of the Integration Editor.
+   */
+  let extensionUrl: string = `/extensions?per_page=50`;
```
Comment (gashcrumb) [inline app/ui-react/packages/api/src/useExtensions.tsx:22]: I wouldn't use backticks here.

File: app/ui-react/packages/api/src/useExtensions.tsx [hunk 3]
```diff
@@ -8,13 +8,28 @@ export interface IExtensionsResponse {
...
+  if (extensionType) {
+    extensionUrl = extensionUrl + `&extensionType=${extensionType}`
```
Comment (gashcrumb) [inline app/ui-react/packages/api/src/useExtensions.tsx:24]: instead I'd suggest:

```suggestion
    extensionUrl = `${extensionUrl}&extensionType=${extensionType}`;
```

File: app/ui-react/packages/ui/stories/Integration/Editor/IntegrationEditorExtensionList.stories.tsx [hunk 4]
```diff
@@ -0,0 +1,68 @@
+import { action } from "@storybook/addon-actions";
+import { text } from "@storybook/addon-knobs";
+import { storiesOf } from '@storybook/react';
+import * as React from 'react';
+
+const stories = storiesOf('Integration/Editor/IntegrationEditorExtensionList', module);
+
+import {
+  IntegrationEditorExtensionTable
+} from '../../../src';
```
Comment (gashcrumb) [inline app/ui-react/packages/ui/stories/Integration/Editor/IntegrationEditorExtensionList.stories.tsx:10]: might be good to just import the one file, I seem to recall someone hitting an issue with an import done this way.

Comment (kahboom) [reply app/ui-react/packages/ui/stories/Integration/Editor/IntegrationEditorExtensionList.stories.tsx:10]: WebStorm complains for the following:
`import { IntegrationEditorExtensionTable } from '../../../src/Integration/Editor/IntegrationEditorExtensionTable';`
but is okay with this:
`import { IntegrationEditorExtensionTable } from '../../../src/Integration/Editor';`
wdyt? I can also ignore it, seems to be the case for some of the other stories.

**Pull request merged**